apps/src/lib/tools/jsinterpreter/JSInterpreter.js

```diff
@@ -245,6 +246,81 @@ export default class JSInterpreter {
     }
   }
 
+  /**
+   * Builds a list of objects that contain all metadata about any functions in
+   * the gived code string. Each object in the returned list has the following
+   * properties:
+   * functionName - the name of the function
+   * parameters - the names of the parameters passed into the function
+   * comment - the comment describing the function. This could be in a JSDoc,
+   * multiline, singleline, or multiple singlelines format.
+   *
+   * @param {string} code - The code to be parsed for functions
+   */
+  static getFunctionsWithComments(code) {
+    let functionsWithComments = [];
+    function getPreviousComment(allComments, startingLocation) {
+      return allComments.find(comment => {
+        return comment.endLocation === startingLocation - 1;
+      });
+    }
+
+    let comments = [];
+    let options = {
+      onComment: (isBlockComment, text, startLocation, endLocation) => {
```

Comment:
How/when is this option used? Is it https://github.com/code-dot-org/JS-Interpreter/blob/master/acorn.js#L81? Might be helpful to include a comment explaining since it's not immediately obvious what this is for.

Reply:
👍  PTAL